### PR TITLE
Display the speaker and connect to it

### LINKED ISSUE #1427: Display the detected smart speaker and connect to its network
**Feature**

The PR #1424 detects the nearby access points and the next step is to display the available devices and then add a button on each displayed device to connect to their WAP.

parent issue #1250

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1427 

The DeviceActivity displays the scanned devices around it and also provides set up option, which simply connects to the Speaker's WAP that is chosen to connect.

### CODE REVIEW

**Comment by batbrain7** [comment]
@Orbiter please review

**Comment by batbrain7** [comment]
@mariobehling please review and merge after this I'll be working on making the api call to the speaker

File: app/src/main/res/layout/activity_device.xml [hunk 1]
```diff
@@ -60,4 +60,12 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:srcCompat="@drawable/ic_add_black_24dp" />
 
+    <android.support.v7.widget.RecyclerView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="4dp"
```
Comment (arundhati24) [inline app/src/main/res/layout/activity_device.xml]: Use dimen resource

File: app/src/main/res/layout/device_layout.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <android.support.v7.widget.CardView
+        android:layout_margin="4dp"
```
Comment (arundhati24) [inline app/src/main/res/layout/device_layout.xml]: dimen resource. There are many such places where hardcoded values need to be replaced by dimen resource.

File: app/src/main/res/layout/device_layout.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            >
+            <ImageView
+                android:layout_width="100dp"
+                android:padding="10dp"
+                android:layout_height="100dp"
```
Comment (arundhati24) [inline app/src/main/res/layout/device_layout.xml]: hardcoded dimens?

File: app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt [hunk 4]
```diff
@@ -70,10 +85,37 @@ class DevicePresenter(deviceActivity: DeviceActivity) : IDevicePresenter {
...
+        SSID = networkSSID
+        deviceView?.showProgress(utilModel.getString(R.string.connecting_device))
+        ConnectWifi().execute()
+    }
+
+    inner class ConnectWifi : AsyncTask<Void, Void, Void>() {
+
+        override fun doInBackground(vararg p0: Void?): Void? {
+            val wifiConfiguration = WifiConfiguration()
+            wifiConfiguration.SSID = "\"" + SSID + "\""
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt]: What if SSID is null?

File: app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt [hunk 5]
```diff
@@ -5,18 +5,30 @@ import org.fossasia.susi.ai.device.contract.IDevicePresenter
...
+/*
+*   Created by batbrain7 on 22/06/18
+*   This is the presenter for the device activity where the logic of connecting to the device is written.
+ */
 
-class DevicePresenter(deviceActivity: DeviceActivity) : IDevicePresenter {
 
+class DevicePresenter(deviceActivity: DeviceActivity, manager: WifiManager) : IDevicePresenter {
+
+    var mWifiManager = manager
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt]: Why not private?

File: app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt (hunk 4 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt]: code won't reach here if it is null

File: app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt (hunk 5 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/device/DevicePresenter.kt]: Also, don't use hungarian notation

**Comment by batbrain7** [comment]
@arundhati24 done the changes, please review now

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -354,5 +354,8 @@
     <string name="digit_five">5</string>
     <string name="location_access">Provide Location Access</string>
     <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby wifi networks available.</string>
-
+    <string name="set_up">SET UP</string>
+    <string name="connecting_device">Connecting to your device...</string>
+    <string name="device_name">SUSI.AI</string>
+    <string name="connection_success">Connected Successfully !</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Remove space after Successfully

**Comment by batbrain7** [comment]
@mariobehling please merge

**Comment by batbrain7** [comment]
@iamareebjamal please review

**Pull request merged**